modules/contrail-alarm-gen.nix

```diff
@@ -0,0 +1,57 @@
+{ config, lib, pkgs, contrailPkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.contrail.alarmGen;
+  confFile = import (./configuration + "/R${contrailPkgs.contrailVersion}/alarm-gen.nix") { inherit pkgs cfg; };
```

Comment:
So here we could either just use `./configuration/common/alarm-gen.nix` or even add some `getConfig` function to that receives `contrailVersion` along with `alarm-gen.nix` and then we'd have the config logic (common or specific for a version) in one place - but maybe that is just over-engineering and a simple static import will be just fine ;-)